docs/adr/0004-fargate-pools.md

```diff
@@ -0,0 +1,45 @@
+# 4. EKS AWSFargatePools
+
+- Status: proposed
+- Date: 2021-02-15
+- Authors: @michaelbeaumont
+- Deciders:
+
+## Context
+
+At the time of writing CAPA has the AWS-specific infrastructure resources `AWSMachinePool`s
```

Comment:
If we are not mapping to a MachinePool then shall we remove 'Pool' from the kind name and go with something like `AWSFargateProfile`?

Reply:
We could. I like the word "pool" better in general but it does have the `MachinePool` connotation. Profile is kind of meaningless IMO, but since that's what the AWS object is called, it wouldn't be wrong.

Reply:
Yeah i see your point....'profile' is a bit meaningless. It is a 'pool' of compute resource to use but it sort of implies MachinePool to me. But if machinepools are changing in v1alpha4 then perhaps it would be better to keep it as is?

Reply:
I think keeping the "Pool" name might get confusing, especially if there is no mapping with `MachinePools`. MachinePools might change but they will still have different objectives and there is no plan to change the name. See https://github.com/kubernetes-sigs/cluster-api/issues/4063 for details.

Reply:
@michaelbeaumont - what do you think could be a good alternative to using Pool or Profile?  Just some ideas: `AWSFargateRule`, `AWSFargatePodExecution`, `AWSFargateExecutionConfig`

Reply:
Let's go with the original suggestion, `AWSFargateProfile`